test/e2e/e2eutil.go

```diff
@@ -194,3 +201,99 @@ func NewTestTLSConfig(optional bool) mdbv1.TLS {
 		},
 	}
 }
+
+func ensureObject(ctx *f.Context, t *testing.T, key types.NamespacedName, obj runtime.Object) {
+	msg := ""
+	if key.Namespace != "" && key.Namespace != "default" {
+		msg = fmt.Sprintf("%s %s/%s", reflect.TypeOf(obj), key.Namespace, key.Name)
+	} else {
+		msg = fmt.Sprintf("%s %s", reflect.TypeOf(obj), key.Name)
+	}
+
+	err := f.Global.Client.Get(context.TODO(), key, obj)
+	if err != nil && !apiErrors.IsNotFound(err) {
```

Comment:
I think this whole `if-else` structure could be simplified

What do you think of a structure like:

```golang
if err != nil {
     if !apiErrors.IsNotFound(err) {
            fatal
     }
    err = f.Global.Client.Create(context.TODO(), obj, &f.CleanupOptions{TestContext: ctx})
		if err != nil {
			t.Fatal(err)
		}
} else {
              fmt.Printf("%s already exists!\n", msg)
		err = f.Global.Client.Update(context.TODO(), obj)
		if err != nil {
			t.Fatal(err)
		}
}
```

I most likely messed up something in coming up with the example, but I hope I managed to explain the idea :)

Reply:
I got the idea! That definitely seems more legible, so I have made the change.